src/main/java/book_store/service/BookServiceImpl.java

```diff
@@ -1,28 +1,41 @@
 package book_store.service;
 
-import book_store.model.Book;
+import book_store.dto.BookDto;
+import book_store.dto.CreateBookRequestDto;
+import book_store.exception.EntityNotFoundException;
+import book_store.mapper.BookMapper;
 import book_store.repository.BookRepository;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
-
+import book_store.model.Book;
 import java.util.List;
+import java.util.Random;
 
+@RequiredArgsConstructor
 @Service
 public class BookServiceImpl implements BookService{
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setIsbn("isbn" + new Random().nextInt(1000));
```

Comment:
random should have been a class field, 1000 - class constant, but you shouldn't generate a random ISBN - it should be a part of request DTO